watertap/unit_models/crystallizer.py

```diff
@@ -558,6 +593,20 @@ def eq_crystallizer_height_constraint(b):
                 a + b + ((a - b) ** 2 + eps**2) ** 0.5
             )
 
+        # 13. Energy available from the vapor
+        @self.Constraint(doc="Thermal energy in the vapor")
+        def eq_vapor_energy_constraint(b):
+            return b.energy_from_vapor == (
+                b.properties_vapor[0].flow_mass_phase_comp["Vap", "H2O"]
+                * (
+                    b.properties_pure_water[
+                        0
+                    ].dh_vap_mass_solvent  # Latent heat from the vapor
```

Comment:
It is still not clear why we need this new stateblock. The only other reason that comes to mind is that it somehow helps with the multi-effect configuration.

Reply:
Same as above.